python/custom_test_files.py

```diff
@@ -0,0 +1,189 @@
+import time
+import yaml
+import shutil as sh
+
+from lar_generator import lar_gen #Imports lar_gen class. 
+lg = lar_gen() #Instantiates lar_gen class as lg. 
+
+class Custom_Test_Files(object):
+    
+    """Returns a Custom_Test_Files object with a set of functions and parameters to create submission files
+    with a specified number of rows."""
+    
+    def __init__(self, filename):
+        self.filename = filename #Instantiates the class with an existing file. 
+        print("Custom_Test_Files Object Instantiated.")
+            
+    def create_file(self, row_count=None, save_file=None):
+        """
+        Creates a new custom file, passing in a row count and filepath to save the created file. 
+    
+        """
+        ts_file = self.separate_ts() #Stores the file path for TS data as ts_file. 
+        lar_file = self.separate_lar() #Stores the file path for LAR data as lar_file. 
+        #Uses the get_rows function to create a dataframe of LAR and to add on the specified number of rows. 
+        df = self.get_rows(row_count=row_count, lar_file=lar_file) 
+        #Uses the assign rows function to create unique ULI's. 
+        df = self.assign_uli(df=df, ts_file=ts_file)
+        #Places the dataframe of LAR in a file with a TS row. 
+        self.lar_to_hmda_file(ts_file=ts_file, df=df, save_file=save_file)
+    
+    def separate_ts(self):  
+        """
+         Separates the Transmittal Sheet (TS) row from the submission file.
+         Saves the TS data as text files in the /edits_files/file_parts/ directory. 
+        
+        """
+        with open(self.filename, 'r+' ) as f: #Opening the submission file. 
+            ts_row = f.readline() #Reading in the first row as the TS submission. 
+            lar_rows = f.readlines() #Reading in the other rows as the LAR submission.  
+        with open("../edits_files/file_parts/ts_data.txt", 'w') as out_ts: #Writes a file of TS data. 
+            out_ts.writelines(ts_row)
+            
+        statement = """ TS file saved as: "../edits_files/file_parts/lar_data.txt" """
+        ts_filepath = "../edits_files/file_parts/ts_data.txt"
+        return ts_filepath #returns a string of the TS filepath.
+    
+    def separate_lar(self):
+        with open(self.filename, 'r+' ) as f: #Opening the submission file. 
+            ts_row = f.readline() #Reading in the first row as the TS submission. 
+            lar_rows = f.readlines() #Reading in the other rows as the LAR submission.  
+        with open("../edits_files/file_parts/lar_data.txt", 'w') as out_lar: #Writes a file of LAR data. 
+            out_lar.writelines(lar_rows)
+        
+        #Prints the location of each data file. 
+        statement = """LAR file saved as: "../edits_files/file_parts/lar_data.txt" """
+        
+        lar_filepath = "../edits_files/file_parts/lar_data.txt"
+        return lar_filepath #returns a string of the LAR filepath.
+    
+    def get_rows(self, lar_file=None, row_count=None):
```

Comment:
The name get_rows is a bit confusing as the docstring says it is returning LAR data and the code is concatenating LAR data to build the return. 

Consider clarifying the function name.